server.go

```diff
@@ -79,6 +93,67 @@ func (srv *Server) Handle(fn Handler) {
 	srv.Handler = fn
 }
 
+// Close immediately closes all active listeners and all active
+// connections.
+//
+// Close returns any error returned from closing the Server's
+// underlying Listener(s).
+func (srv *Server) Close() error {
+	srv.mu.Lock()
+	defer srv.mu.Unlock()
+	srv.closeDoneChanLocked()
+	err := srv.closeListenersLocked()
+	for c := range srv.activeConn {
+		c.Close()
+		delete(srv.activeConn, c)
+	}
+	return err
+}
+
+// shutdownPollInterval is how often we poll for quiescence
+// during Server.Shutdown. This is lower during tests, to
+// speed up tests.
+// Ideally we could find a solution that doesn't involve polling,
```

Comment:
It's not clear to me why polling is necessary? Are we waiting for all of the clients to successfully disconnect before Close() returns? Is this a requirement?

In my code, I typically disable the server from accepting new connections and then in separate goroutine(s) I start issuing disconnect orders/closing connections while the original goroutine just returns.